openshift/templates/nats.yaml

```diff
@@ -0,0 +1,208 @@
+kind: Template
+metadata:
+  name: nats
+  annotations:
+    "openshift.io/display-name": nats
+parameters:
+  - name: APP_NAME
+    description: Application name (wps - wildfire predictive services)
+    value: wps
+  - name: SUFFIX
+    description: Deployment suffix, e.g. pr-###
+    required: true
+objects:
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: nats-config
+      labels:
+        app: nats
+    data:
+      nats.conf: |-
+        pid_file: "/var/run/nats/nats.pid"
+        http: 8222
+        max_payload: 100Mb
+        max_pending: 120Mb
+        debug: false
+        trace: false
+        trace_verbose: false
+        server_name: $POD_NAME
+        cluster {
+            port: 6222
+            name: wps-nats-cluster
+            routes [
+               nats://nats-0.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+               nats://nats-1.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+               nats://nats-2.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+               nats://nats-3.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+               nats://nats-4.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+            ]
+            cluster_advertise: $CLUSTER_ADVERTISE
+            connect_retries: 50
+        }
+        leafnodes {
+          port: 7422
+        }
+        jetstream {
+          store_dir: /data
+          max_file: 5Gi
+        }
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      name: nats
+      labels:
+        app: nats
+    spec:
+      selector:
+        app: nats
+      clusterIP: None
+      ports:
+        - name: client
+          port: 4222
+        - name: cluster
+          port: 6222
+        - name: monitor
+          port: 8222
+        - name: metrics
+          port: 7777
+        - name: leafnodes
+          port: 7422
+        - name: gateways
+          port: 7522
+
+  - apiVersion: apps/v1
+    kind: StatefulSet
+    metadata:
+      name: nats
+      labels:
+        app: nats
+    spec:
+      selector:
+        matchLabels:
+          app: nats
+      replicas: 5
+      serviceName: "nats"
+      volumeClaimTemplates:
+        - metadata:
+            name: wps-nats-jetstream
+          spec:
+            accessModes:
+              - ReadWriteMany
+            volumeMode: "Filesystem"
+            persistentVolumeReclaimPolicy: Delete
+            resources:
+              requests:
+                storage: 5Gi
+      template:
+        metadata:
+          labels:
+            app: nats
+        spec:
+          # Common volumes for the containers
+          volumes:
+            - name: config-volume
+              configMap:
+                name: nats-config
+            - name: pid
+              emptyDir: {}
+            - name: wps-nats-jetstream
+              persistentVolumeClaim:
+                claimName: wps-nats-jetstream
+
+          # Required to be able to HUP signal and apply config reload
```

Comment:
what's HUP mean?

Reply:
https://en.wikipedia.org/wiki/SIGHUP

Sorry most of these comments were copy-pasta'd from: https://github.com/bcgov/EDUC-INFRA-COMMON/blob/master/openshift/nats/nats.yml